atc/db/pipeline.go

```diff
@@ -621,6 +631,45 @@ func (p *pipeline) Unpause() error {
 	return tx.Commit()
 }
 
+func (p *pipeline) Archive() error {
+	tx, err := p.conn.Begin()
+	if err != nil {
+		return err
+	}
+
+	defer Rollback(tx)
+
+	_, err = psql.Update("pipelines").
+		Set("archived", true).
+		Set("paused", true).
+		Where(sq.Eq{
+			"id": p.id,
+		}).
+		RunWith(tx).
+		Exec()
+
+	if err != nil {
+		return err
+	}
+
+	err = p.clearConfigForJobsInPipeline(tx)
```

Comment:
Might be important to point out in the notes that those who go through a down migration will end up with pipelines that, if unpaused, won't really work until the `set-pipeline` as failures like

```
web_1     | {"timestamp":"2020-04-09T15:29:11.132091560Z","level":"info","source":"atc","message":"atc.checker.errored","data":{"check":2,"error":"run check step: check: resource script '/opt/resource/check []' failed: exit status 1\n\nstderr:\ninvalid configuration: must configure either 'interval' or 'start' and 'stop'\n","session":"18"}}
```

would occur :thinking:  wdyt?

Reply:
lol, after doing a down migration I just realized that a pipeline without resources like this actually works

```
jobs:
- name: ""
  plan: null
```

(you can get a successful build :thinking: not sure if the `name: ""` is really something we should accept when scheduling the build, @vito's https://github.com/concourse/concourse/issues/3985 would go against it)

---

if you're curious about the events of such build:

```
fly -t local curl /api/v1/builds/3/events
id: 0
event: event
data: {"data":{"status":"started","time":1586446873},"event":"status","version":"1.0"}

id: 1
event: event
data: {"data":{"status":"succeeded","time":1586446893},"event":"status","version":"1.0"}

id: 2
event: end
data
```

Reply:
https://github.com/concourse/concourse/pull/5387/commits/6ea4c55ba0fb56520454a76ec90f8e94ad56a4d0